### PR TITLE
Import fixes for sanitizer reported issues

### PR DESCRIPTION
**Pull request by fanquake**
See discussion in https://github.com/bitcoin/bitcoin/pull/22646. 

Note that this code takes advantage of C++11 features, however this is fine, because this tree is for our downstream use, where we already require C++17.

Co-authored-by: MarcoFalke <falke.marco@gmail.com>

### CODE REVIEW

**Comment by maflcko** [comment]
cr ACK 7d3b9756c133c2e08570d9fb514b00e90bee857a

**Comment by practicalswift** [comment]
Concept ACK

File: lib/univalue.cpp [hunk 1]
```diff
@@ -178,17 +178,19 @@ bool UniValue::findKey(const std::string& key, size_t& retIdx) const
...
 bool UniValue::checkObject(const std::map<std::string,UniValue::VType>& t) const
 {
-    if (typ != VOBJ)
+    if (typ != VOBJ) {
         return false;
+    }
 
-    for (std::map<std::string,UniValue::VType>::const_iterator it = t.begin();
-         it != t.end(); ++it) {
+    for (const auto object: t) {
```
Comment (maflcko) [inline lib/univalue.cpp]: Missing `&`?

**Comment by maflcko** [comment]
Concept ACK 24ba142c150e2bec21f345503b7ea053f0e222dc 🦌

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Concept ACK 24ba142c150e2bec21f345503b7ea053f0e222dc 🦌
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhDPAwAq1ucIUxikX4ULnn1HFJZlRizyyaukXhXZo2RXswRVsSNixTO26udaIEA
9Cr4fo2Rjb+rd8nPkNF91mRQ0yQlfO25I6gq04ba0Ncm2nfKrXjc5XukRQPpheYQ
yR1YHyU48s24MB687BHlmEWn6saKnOyyERqgcpnxalLH9i0iOf/MpHjci8yHv2z4
yvLm7wO/bJRFvgXWhZTkjmhLUo1UTkexG+TOiiln4yimRkjFpGL/5UFGp7YDIlbj
qdUQN+ahelCR2T6VWRUPNAm6EQJ4YypedYUOqV3GNGGGM5nUeKq3jkEzeM1Ns3CI
RPi+dz/ulAQaPzfIs4fr/g2KpPNhaWvJw65KO3MLHzXyzhv9E86xj7Pg6kTeSsEa
EwKt4vu553UCdcvwbIzByy8NCKQ7Ac1A5/Xu/xokOSrPknu2yvc8CejjFogdKtTg
TtSqjFB/jAtGcHSPx5NqTMhMg5YQ6zR7qwSod35eRNCSAdZkRvZm+9U+p2yFNUtk
5qYIVga2
=hz7z
-----END PGP SIGNATURE-----
```

</details>

File: lib/univalue.cpp (hunk 1 above)
Comment (fanquake) [reply lib/univalue.cpp]: Done

**Comment by maflcko** [comment]
Concept ACK 135254331e6fa410e7e79f947851ee37af67543f 🚍

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Concept ACK 135254331e6fa410e7e79f947851ee37af67543f 🚍
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUiosAv/aS5IarJDUr+zkZUfcwrkRg2I38fHvA7Jtdg1/NDd6J7WzzEy/m3Kjp3R
/CidE2qZwvtfjqkgYwbSTggbcrysrQ7r5NhlQs0FZa2O/higPRXuObGYQAX7bNir
FM5/aKcW6jK83ruMFfSv8bunJ8slMD3mfX8FSVzh2LAoNHDHwFiz1257OfT1BLs9
zSttgAw3mU5T5yoXwqknnBb0aWPsY83NSnlyRN+Wk8Hiu0BZdCKmQ6JJLYiFt2IZ
o4wi1hDT5edHu1GOQBJTtT6xyRsaLugc0txlNCLEKafv7167XiLajTDAFg8gQ9/o
Dv6QpDOHBqU2gWysz7gNr95mT/rBbRvXOZkYP8UPdrS3Sgj+vLB49ZsDEQoksW9B
4Fhxj8zJLEMZpImlTgmBkZEn3UO9W2Ss5fuZaCs6ZgbZ1gvb+OYUXxuBnZYlcnW7
4kehqWhKet83nwVdP9udl3tax5bBRS7znCri26Ndv7tZ5ctRUdChCsXu3zjIed5m
WCDQVAYz
=NmZ3
-----END PGP SIGNATURE-----
```




</details>

**Comment by laanwj** [comment]
Post-merge ACK

**Pull request merged**